Libraries/Text/RCTTextField.m

```diff
@@ -42,6 +43,14 @@ - (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher
 
 - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
 {
+  // If the range length is 1, it signifies a backspace key.
+  // We should ignore them here and let the `deleteBackward` method handle all backspace keys.
+  if (range.length == 1) {
```

Comment:
Won't this also detect pasting over single character selections?  Will that cause weird things to happen?